src/image.rs

```diff
@@ -282,13 +282,19 @@ impl ImageClient {
             ));
         }
 
+        let mut layers = Vec::new();
```

Comment:
Maybe use `unique_layers` will be more straightforward

Reply:
Thanks, much clearer! fixed it